server/src/main/java/org/opensearch/cluster/ClusterModule.java

```diff
@@ -373,6 +375,9 @@ public static Collection<AllocationDecider> createAllocationDeciders(
         addAllocationDecider(deciders, new AwarenessAllocationDecider(settings, clusterSettings));
         addAllocationDecider(deciders, new NodeLoadAwareAllocationDecider(settings, clusterSettings));
         addAllocationDecider(deciders, new TargetPoolAllocationDecider());
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE_MIGRATION_EXPERIMENTAL_SETTING)) {
```

Comment:
Would prefer Feature flag PR as a separate PR such that all other code changes can integrate without further delay

Reply:
Sure, would separate it out of this PR

Reply:
Got it now. Added back again.